src/firecracker/src/main.rs

```diff
@@ -311,8 +311,8 @@ fn main_exitable() -> ExitCode {
         .map(fs::read_to_string)
         .map(|x| x.expect("Unable to open or read from the mmds content file"));
 
-    if let Some(data) = metadata_json {
-        MMDS.lock()
+    if let (Some(data), Some(mmds)) = (metadata_json, MMDS.as_ref()) {
```

Comment:
This change is good, but we should also disallow the usage of `--metadata` when we run with `--no-api`, since MMDS will be disabled.